model/rulefmt/rulefmt.go

```diff
@@ -110,6 +110,20 @@ func (g *RuleGroups) Validate(node ruleGroups) (errs []error) {
 			)
 		}
 
+		for k, v := range g.Labels {
+			if !model.LabelName(k).IsValid() || k == model.MetricNameLabel {
```

Comment:
Does Prometheus do this validation for per-rule labels?  If not, do we want to be inconsistent?
Also, should we not do the same validation from `promtool check rules` ?

Reply:
It's being called by `promtool check rules` here: https://github.com/prometheus/prometheus/blob/main/cmd/promtool/main.go#L808

Reply:
Could you answer this too please?
> Does Prometheus do this validation for per-rule labels? If not, do we want to be inconsistent?

Reply:
ping

Reply:
Take it step by step:
You are adding a validation?
You are adding it in one place, which is similar to another place?
It would be inconsistent to add it in one place and not the other place?

Reply:
Well for per rule validation it's being called here from here (https://github.com/prometheus/prometheus/blob/main/model/rulefmt/rulefmt.go#L115-L116) and then validate each label per rule here (https://github.com/prometheus/prometheus/blob/main/model/rulefmt/rulefmt.go#L221-L233). So I'm not sure how it would be inconsistent.

Reply:
ping @bboreham

Reply:
ping @bboreham